californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java

```diff
@@ -862,4 +862,16 @@ public static Request newPut() {
 	public static Request newDelete() {
 		return new Request(Code.DELETE);
 	}
+
+	/**
+	 * Seta a new value for the request code.
+	 * 
+	 * Needed for obfuscation in OSCORE.
+	 * 
+	 * @param value
+	 */
+    public void setCode(Code value) {
+       this.code = value;
+        
+    }
```

Comment:
FMPOV, the current Request/Reponse/Message is pretty unclear about what could be set/changed in which stage of processing the message. With adding this setter, it get's even more unclear. 

`public static byte[] decrypt(Request request) `

seems to change so much of the request, that I would try to go for creating a new request (forwarding/copying the MessageObserver) instead of exchanging that much of the request.

Reply:
- But then why don't make an argument to remove more of the class' setters? (I followed the class' already established setters/getters).
- The current way is obviously more convenient, but I could just do as you say if it feels/looks better for the code.

Reply:
The current Message API design is not mine :-). 

But I got aware, that it's hard to know, what is when possible and what will fail, when called at the wrong time. Therefore I have my doubt, if it is a good idea to add that setter.